### DISCUSSION TITLE
CLaM Meeting Discussion Ideas

### DISCUSSION THREAD

**Discussion by BenHenning** [oppia/oppia-android#5602] [General Discussion]
# Overview & instructions

Starting in 2025, the Core Learner and Mastery team is adjusting its meeting format to be more collaborative and widely applicable to all members of the team. This entails two main things:
1. Primary discussions in the weekly meeting will encompass topics of general interest such as show-and-tells, tech talks, announcements, and more.
2. Discussions to debug or work through a specific problem people are facing will be done in a follow-up weekly "office hours" with the team leads.

This comment tracks all potential discussion topics for upcoming CLaM meetings, and the corresponding person who will conduct them. We aim to have 1-2 of these each week. If you're interested in discussing topics that you'd like to see or would like to contribute, please reply to this discussion thread.

# Plans for the next CLaM meeting

The current topics planned for the next CLaM meetings (April 2025) are:
- TBD

# Future discussion topics

The different types of discussion topics include:
- **Show-and-tells**: these are quick demos meant to simply show what someone is working on. They can include 1 or 2 areas that the presenter is looking for feedback on,but this isn't a requirement. This is also _not_ a formal review of ongoing work--the point is simply to share progress with the team. :)
- **Informal technical discussions**: these are (mostly) unplanned discussions around a particular idea or solution. This isn't a time to debug a particular issue, nor is it a fully fledged tech talk. There is a primary person who may have a list of subtopics to discuss, but they're essentially providing an open discussion prompt for the team.
- **Tech talks**: these are more detailed talks meant to teach something in depth. They should include a slide deck and very likely will be recorded so that other members of the team (including future ones) can benefit from them.
- **Announcements**: these are significant team/Oppia changes or milestones that are useful for helping to continue building context of the CLaM team's work among the broader Oppia and education ecosystem.
- **Bug break-down**: these are collaborative work sessions where we go through a single issue and work to disambiguate it by digging into the potential root causes and potential solutions. The goal here is to essentially construct starter issues from less detailed bug reports or other issues to help onboard new team members.
- **Games**: not exactly a discussion prompt, but it's something fun to do together.

## Show-and-tells

- [x] New profile selection list changes being done as part of the onboarding v2 project by @adhiamboperes.
- [x] Setting up Oppia Android on Docker by @whyash8 
- [x] Demo of #5600 by @subhajitxyz 
- [x] Setting up the Bazel Plugin by @BenHenning 
- [x] The HTML Parser by @manas-yu  Slides: https://docs.google.com/presentation/d/1ytfK4LZ0hZhcA-i4iiwSt1aigXz1XU-FQtaEjgmhess/edit?usp=drivesdk

## Tech talks

- [ ] How to give both formal and informal tech talks.

## Informal technical discussions

- [x] Discuss how to debug GitHub Actions by @Rd4dev
- [x] Testing at Oppia Android by @adhiamboperes 
- [x] Triaging and prioritizing issues for the CLaM team by @adhiamboperes 
- [x] Writing design docs by @adhiamboperes 

## Announcements

None planned yet!

## Bug break-down sessions

- [x] [Drag & drop is slow to respond on slow devices](https://github.com/oppia/oppia-android/issues/2974#issuecomment-2571385330)
- [x] [Bazel Linter Integration](https://github.com/oppia/oppia/issues/3000)
- [x] [Content and dark mode issues](https://github.com/oppia/oppia-android/issues?q=is%3Aissue%20state%3Aopen%20author%3Acam-pinheiro%20%20label%3A%22Impact%3A%20High%22%20%20) 

## Games

None requested yet!

**Comment by TanishMoral11** [comment]
Hey @BenHenning , 
Could you please add issue #3000 to the list of discussion topics? It involves setting up Buildifier with GitHub checks and pre-push hooks, and could be relevant for an upcoming meeting. 
Thank you!

**Reply by BenHenning** [reply]
Thanks for bringing this up @TanishMoral11. It's actually less important than it seems. We already integrate with buildifier. #3000 is simply recommending that we move the check into Bazel proper rather than using ``static_checks.sh`` (which is also true for other checks like ktlint, buf, and Java checkstyle). I have a PR I started a long time ago that introduces a single replacement for static checks (with progress checking and nice colorizing) that should address #3000, but fortunately it isn't urgent. :)

**Reply by BenHenning** [reply]
To close the loop on this, the bit about moving Buildifier support into a Kotlin script has now been defined in #4106 (as part of updating other similar scripts) and #3000 has now been closed with an update to the pre-push hook.

**Comment by manas-yu** [comment]
I wanted to provide an update on this issue (#4996) following this [comment](https://github.com/oppia/oppia-android/issues/4996#issuecomment-2572648335). I’ve identified and implemented a solution that resolves the build errors caused by the default Windows encoding, and the build is now working fine.

Given that we’re in the Gradle phase-out stage, I’d like to discuss whether we should proceed with these changes or hold off. Should this be prioritized, or would it make sense to defer?

Additionally, while this solution addresses the immediate problem, I believe implementing a .gitattributes file remains important. It’s a good practice for ensuring consistent encoding and line-ending handling across the repository, which will benefit contributors in diverse environments.

**Reply by BenHenning** [reply]
I agree with your point that it's still important, but likely less urgent to implement due to Gradle no longer being an option for development.

**Comment by manas-yu** [comment]
@BenHenning, 

As per the discussion with @adhiamboperes in the meeting, I'm updating it here for further discussion on PR #5677. This PR introduces the `OnItemTouchListener` to handle the long press duration and reduce it via `onInterceptTouchEvent`. 

The implementation works well in manual testing, but the tests are failing. Previously, the long press time was around **500ms** by default, as implemented in `ItemTouchHelper.SimpleCallback` for all devices. With `onInterceptTouchEvent`, we were able to control it significantly. 

The failing tests might be related to issue #1612. Further insights on the potential cause and resolution would be helpful. Thanks!

**Comment by adhiamboperes** [comment]
@adhiamboperes proposed a topic: Estimating project timelines

**Comment by adhiamboperes** [comment]
@manas-yu proposed a tech talk: InMemoryBlockingCache

**Comment by adhiamboperes** [comment]
@subhajitxyz proposed discussions on:
1. Advanced parts of the codebase and how they contribute to the overall architecture. 
2. Future development plans, and potential improvements.

**Reply by adhiamboperes** [reply]
It would be great to do a breakdown of which sections are of interest to different folks.

**Comment by adhiamboperes** [comment]
@whyash8 proposed a talk on advanced architectural patterns in android.

**Reply by adhiamboperes** [reply]
Plan: To list the different patterns of interest and schedule them for discussion seperately.

**Comment by adhiamboperes** [comment]
**Tue 26th May, 2026** - Live code reviews

**Comment by adhiamboperes** [comment]
**Unscheduled** - Using AI coding Agents effectively
